Makefile

```diff
@@ -28,17 +60,43 @@ LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 |  awk '{print $$7}')
 
 # Can choose different docker versions see list here - https://hub.docker.com/_/docker/
 HOST_CHECKOUT_DIR?=$(CURDIR)
-CONTAINER_NAME?=calico/libnetwork-plugin$(ARCHTAG)
-GO_BUILD_CONTAINER?=calico/go-build$(ARCHTAG):$(GO_BUILD_VER)
+CONTAINER_NAME?=calico/libnetwork-plugin:latest-$(ARCH)
 DIST=dist/$(ARCH)
 PLUGIN_LOCATION?=$(CURDIR)/$(DIST)/libnetwork-plugin
```

Comment:
Consistent naming: In the other builds, we have used a single `dist/` or `bin/` directory (probably should unify that too, but doesn't matter all that much), with the output files named `dist/<filename>-<arch>`, e.g. `felix-amd64` or `typha-ppc64le`, whereas this one does `dist/<arch>/<filename>`, e.g. `dist/amd64/libnetwork-plugin`. 

I think we should take advantage of this cleanup to unify the filename, make the outputs `dist/libnetwork-plugin-<arch>`.

It would mean changes to the `Makefile` and the `Dockerfile.*`, but that is it.